scripts/sidebar-macros.js

```diff
@@ -108,4 +108,22 @@ class MacroSidebarDirectory extends SidebarDirectory {
 
     /** @override */
     static documentName = "Macro";
-}
\ No newline at end of file
+}
+
+MacroSidebarDirectory.prototype._getEntryContextOptions = function newMacroContext() {
+    const options = SidebarDirectory.prototype._getEntryContextOptions.call(this);
+    return [
+        {
+            name: "Execute",
+            icon: `<i class="fas fa-terminal"></i>`,
+            condition: li => {
+                const macro = game.macros.get(li.data("entityId"));
+                return macro.data.img !== CONST.DEFAULT_TOKEN;
```

Comment:
I'm not super familiar with Javascript, and I mostly cobbled this together from similar modules. Thus, I have absolutely no idea what this line does. 😳